Data_Engine/Create/Table.cs

```diff
@@ -0,0 +1,139 @@
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using System;
+using System.Data;
+using System.Linq;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.ComponentModel;
+using BH.oM.Reflection.Attributes;
+using BH.Engine.Reflection;
+using BH.oM.Base;
+using BH.oM.Data.Collections;
+
+namespace BH.Engine.Data
+{
+    public static partial class Create
+    {
+        /***************************************************/
+        /**** Public Methods                            ****/
+        /***************************************************/
+
+        public static Table Table(string name, List<IBHoMObject> bhomObjects, bool ignoreName = false, bool ignoreGuid = true, bool ignoreTags = true)
+        {
+            IBHoMObject first = bhomObjects.First();
+
+            Dictionary<string, object> properties = first.PropertyDictionary();
+
+            List<DataColumn> columns = new List<DataColumn>();
+
+            properties.CleanUnwantedProperties(ignoreName, ignoreGuid, ignoreTags);
+
+            foreach (var kvp in properties)
+            {
+                
+                columns.Add(new DataColumn(kvp.Key, kvp.Value.GetType()));
+            }
+
+            DataTable table = new DataTable();
+
+            table.Columns.AddRange(columns.ToArray());
+
+            foreach (IBHoMObject obj in bhomObjects)
+            {
+                Dictionary<string, object> props = obj.PropertyDictionary();
+                props.CleanUnwantedProperties(ignoreName, ignoreGuid, ignoreTags);
+
+                DataRow row = table.NewRow();
+
+                foreach (var kvp in props)
+                {
+                    row[kvp.Key] = kvp.Value;
+                }
+                table.Rows.Add(row);
+            }
+
+            return new Table { Data = table, Name = name };
+        }
+
+        /***************************************************/
+
+        private static void CleanUnwantedProperties(this Dictionary<string, object> dict, bool ignoreName, bool ignoreGuid, bool ignoreTags)
+        {
+            if (ignoreName && dict.ContainsKey("Name"))
+                dict.Remove("Name");
+            if (ignoreGuid && dict.ContainsKey("BHoM_Guid"))
+                dict.Remove("BHoM_Guid");
+            if (ignoreTags && dict.ContainsKey("Tags"))
+                dict.Remove("Tags");
+
+        }
+
+        /***************************************************/
+
+        [Description("")]
+        [Input("", "")]
+        [Output("", "")]
+        public static Table Table<T>(string name, string axis1Name, List<IComparable> axis1Values, string axis2Name, List<IComparable> axis2Values, List<List<T>> values, string valuesName = "Values")
```

Comment:
What is an `IComparable` exactly? Can we input it from the UI side of things?
Not commenting all the rest, it's in other places as well.